server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -98,6 +98,8 @@
 
     public static final String SYNC_COMMIT_ID = "sync_id";
     public static final String HISTORY_UUID_KEY = "history_uuid";
+    // The default number of hits that one search should return when reading Lucene history
+    public static final int LUCENE_HISTORY_DEFAULT_SEARCH_BATCH_SIZE = 1024;
```

Comment:
can you move this to LuceneChangesSnapshot ? this will allow using a short name like `DEFAULT_BATCH_SIZE`